hibernate-core/src/main/java/org/hibernate/engine/profile/FetchProfile.java

```diff
@@ -111,19 +111,6 @@ public void addFetch(final Fetch fetch) {
 						return;
 					}
 				}
-
-				// also, in cases where we are asked to add a collection join
-				// fetch where we had already added a bag join fetch previously,
-				// we need to go back and ignore that previous bag join fetch.
-				if ( containsJoinFetchedBag ) {
-					// just for safety...
-					if ( fetches.remove( bagJoinFetch.getAssociation().getRole() ) != bagJoinFetch ) {
-						LOG.unableToRemoveBagJoinFetch();
-					}
-					bagJoinFetch = null;
-					containsJoinFetchedBag = false;
-				}
-
```

Comment:
IIRC i removed this because `containsJoinFetchedBag` is never assigned, therefore always by default false, hence never going to enter that if anyway. Following the same reasoning `containsJoinFetchedBag` is also useless.

Reply:
Please revert that, let's not add unrelated changes. The `isContainsJoinFetchedBag()` method where this field is used might be deprecated, but who know what is using this class.